### PR TITLE
GG-29950 Not optimal index is used for distributed query with ORDER BY 

### PR DESCRIPTION
**Pull request by tledkov**

### CODE REVIEW

**Review by AMashenkov — APPROVED** [review]

File: modules/h2/src/main/java/org/h2/result/SortOrder.java [hunk 1]
```diff
@@ -92,6 +92,8 @@
      */
     private final ArrayList<SelectOrderBy> orderList;
 
+    private final Query qry;
```
Comment (korlov42) [inline modules/h2/src/main/java/org/h2/result/SortOrder.java:95]: javadoc

File: modules/h2/src/main/java/org/h2/result/SortOrder.java [hunk 2]
```diff
@@ -101,11 +103,12 @@
      * @param orderList the original query order list (if this is a query)
      */
     public SortOrder(Database database, int[] queryColumnIndexes,
-            int[] sortType, ArrayList<SelectOrderBy> orderList) {
+            int[] sortType, ArrayList<SelectOrderBy> orderList, Query qry) {
```
Comment (korlov42) [inline modules/h2/src/main/java/org/h2/result/SortOrder.java:106]: please add new param to javadoc

File: modules/h2/src/test/java/org/h2/test/synth/TestChooseIndexWithOrderBy.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+import java.util.List;
+import org.h2.util.StringUtils;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * A test that runs random join statements against two databases and compares the results.
```
Comment (korlov42) [inline modules/h2/src/test/java/org/h2/test/synth/TestChooseIndexWithOrderBy.java:25]: misleading javadoc

File: modules/h2/src/test/java/org/h2/test/synth/TestChooseIndexWithOrderBy.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+     */
+    @AfterClass
+    public static void cleanup() throws SQLException {
+        sql("DROP TABLE IF EXISTS TEST");
+
+        connection.close();
+    }
+
+    /**
+     * Check query plan. HASH_JOIN_IDX index must be chosen.
```
Comment (korlov42) [inline modules/h2/src/test/java/org/h2/test/synth/TestChooseIndexWithOrderBy.java:65]: misleading javadoc

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/BasicSqlTest.java [hunk 5]
```diff
@@ -70,6 +70,40 @@ public void testSplitCastFunctionInSortExpression() {
         assertEquals("Emma", res.get(2).get(0));
     }
 
+    /**
+     */
```
Comment (korlov42) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/BasicSqlTest.java:74]: test description

File: modules/h2/src/test/java/org/h2/test/synth/TestChooseIndexWithOrderBy.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        assertTrue("Unexpected plan: " + plan, plan.contains("IDX_VAL0_VAL1"));
+    }
+
+    /**
+     * @param sql SQL query.
+     * @param params Parameters.
+     * @throws SQLException On error.
+     * @return Result set or updated count are printed to string.
+     */
+    private static String sqlStr(String sql, Object... params) throws SQLException {
```
Comment (Berkof) [inline modules/h2/src/test/java/org/h2/test/synth/TestChooseIndexWithOrderBy.java:89]: Can we move sqlStr to some utils class? We have exactly the same code in TestHashJoin class and pretty the same in many others sql tests.

**Review by Berkof — APPROVED** [review]

**Pull request closed without merging**